resources/qml/PrintMonitor.qml

```diff
@@ -20,49 +20,462 @@ Column
+            }
+            MouseArea //Change cursor on hovering.
+            {
+                id: mouseArea
+                hoverEnabled: true
+                anchors.fill: parent
+                cursorShape: Qt.IBeamCursor
+            }
+            TextInput
+            {
+                id: preheatTemperatureInput
+                font: UM.Theme.getFont("default")
+                color: UM.Theme.getColor("setting_control_text")
+                selectByMouse: true
+                maximumLength: 10
+                validator: RegExpValidator { regExp: /^-?[0-9]{0,9}[.,]?[0-9]{0,10}$/ } //Floating point regex.
+                anchors.left: parent.left
+                anchors.leftMargin: UM.Theme.getSize("setting_unit_margin").width
+                anchors.right: parent.right
+                anchors.verticalCenter: parent.verticalCenter
+
+                Binding
+                {
+                    target: preheatTemperatureInput
+                    property: "text"
+                    value:
+                    {
+                        // Stacklevels
+                        // 0: user  -> unsaved change
+                        // 1: quality changes  -> saved change
+                        // 2: quality
+                        // 3: material  -> user changed material in materialspage
+                        // 4: variant
+                        // 5: machine_changes
+                        // 6: machine
+                        if ((bedTemperature.resolve != "None" && bedTemperature.resolve) && (bedTemperature.stackLevels[0] != 0) && (bedTemperature.stackLevels[0] != 1))
+                        {
+                            // We have a resolve function. Indicates that the setting is not settable per extruder and that
+                            // we have to choose between the resolved value (default) and the global value
+                            // (if user has explicitly set this).
+                            return bedTemperature.resolve;
+                        }
+                        else
+                        {
+                            return bedTemperature.properties.value;
+                        }
+                    }
+                    when: !preheatTemperatureInput.activeFocus
+                }
+            }
+        }
+
+        UM.RecolorImage
+        {
+            id: preheatCountdownIcon
+            width: UM.Theme.getSize("save_button_specs_icons").width
+            height: UM.Theme.getSize("save_button_specs_icons").height
+            sourceSize.width: width
+            sourceSize.height: height
+            color: UM.Theme.getColor("text")
+            visible: preheatCountdown.visible
+            source: UM.Theme.getIcon("print_time")
+            anchors.right: preheatCountdown.left
+            anchors.rightMargin: UM.Theme.getSize("default_margin").width / 2
+            anchors.verticalCenter: preheatCountdown.verticalCenter
+        }
+
+        Timer
```

Comment:
I think it is better to put this timer into the actual print device. That way the pre-heat timer becomes a device specific thing and it would become possible to have multiple devices pre-heating. In addition, it makes things a lot simpler when (if?) the machine exposes the pre-heat timer.